Add uploadBuild endpoint

devbdavis opened the pull request:
This API allows you to store build information in artifactory.

Commit:
Add uploadBuild endpoint

cgadam commented on lib/ArtifactoryAPI.js:
```diff
@@ -271,4 +272,39 @@ ArtifactoryAPI.prototype.downloadFile = function (repoKey, remotefilePath, desti
+  var deferred = Q.defer();
+  if(!buildInfo.name || !buildInfo.number) {
+    deferred.reject('Build Info must include a name and number. See https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-BuildUpload for more info');
+  }
```
You need to return the deferred object here, otherwise the other lines below this will be called (particularly request.put which I am sure you don't want to call.

`
deferred.reject('reason');
return deferred.promise;
`

devbdavis replied:
Thank you. I will take a look at this. A return makes sense

cgadam commented on lib/ArtifactoryAPI.js:
```diff
@@ -271,4 +272,39 @@ ArtifactoryAPI.prototype.downloadFile = function (repoKey, remotefilePath, desti
+ */
+ArtifactoryAPI.prototype.uploadBuild = function (buildInfo) {
+  var deferred = Q.defer();
+  if(!buildInfo.name || !buildInfo.number) {
```
Both number are name seems to be strings, right? Then I guess you could call trim() just to discard that someone is sending a bunch of spaces.

devbdavis replied:
This is a good point. I believe artifactory will allow blank string? But that is probably not true and really bad practice. I will add some trims :)

devbdavis commented:
@cgadam I believe I have addressed your concerns on this one. If possible can you please take another look =) Let me know if anything else pops out.

Once these PRs go through, I'm wondering if a refactor could be a good idea. And maybe some tests =)
